src/Util/Singleton.hpp

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (C) 2019-2020 Ashar Khan <ashar786khan@gmail.com>
+ *
+ * This file is part of CP Editor.
+ *
+ * CP Editor is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * I will not be responsible if CP Editor behaves in unexpected way and
+ * causes your ratings to go down and or lose any important contest.
+ *
+ * Believe Software is "Software" and it isn't immune to bugs.
+ *
+ */
+
+// This class is from https://stackoverflow.com/a/46178158
```

Comment:
You don't really need to mention it since it's a general Curiously Recurring template pattern for static polymorphism. If you have copy pasted the code, you can keep it.